src/backend/distributed/test/system_utils.c

```diff
@@ -0,0 +1,132 @@
+/*-------------------------------------------------------------------------
+ *
+ * system_utils.c
+ *	  Methods for managing databases during regression tests.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ * $Id$
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+#include "libpq-fe.h"
+#include "pgstat.h"
+#include "port.h"
+#include "storage/fd.h"
+
+#include "distributed/version_compat.h"
+
+PG_FUNCTION_INFO_V1(citus_stop_test_worker);
+PG_FUNCTION_INFO_V1(citus_start_test_worker);
+
+/*
+ * Invokes pg_ctl stop for the worker using the given port.
+ */
+Datum
+citus_stop_test_worker(PG_FUNCTION_ARGS)
+{
+	int port = PG_GETARG_INT32(0);
```

Comment:
These functions are super-powerful in the sense that they can try to stop/start a database. And the Citus binaries would include these functions, so anyone can create them. 

Is it possible that we error out when called outside regression tests? 

We at least should prevent non-supers.

Reply:
Updated to prevent non-supers. Not anyone can create these functions, as only superusers are allowed to `CREATE FUNCTION LANGUAGE C`